src/main/java/duke/Parser.java

```diff
@@ -0,0 +1,130 @@
+package duke;
+import java.time.LocalDate;
+import java.time.format.DateTimeParseException;
+import java.util.ArrayList;
+
+/**
+ *  A class that handles parsing and processing of user inputs.
+ *  @author  Chen Guanzhou
+ *  @version v1
+ */
+public class Parser {
+    private TaskList currList;
```

Comment:
Perhaps it would be more appropriate if tasklist was not an attribute of parser as a parser does not have a tasklist ?🤔

Reply:
Good point with the has-a relationship, will think about changing this